src/middlewared/middlewared/plugins/account.py

```diff
@@ -229,8 +235,13 @@ class Config:
                             errno.EEXIST
                         )
 
-                    # If it exists, ensure the user is owner
-                    os.chown(data['home'], data['uid'], group['gid'])
+                    # If it exists, ensure the user is owner.
+                    await self.middleware.call('filesystem.chown', {
+                        'path': data['home'],
+                        'uid': data['uid'],
+                        'gid': group['gid'],
+                        'options': {'recursive': True}
```

Comment:
recursive=True is both is a change of behavior, are we sure we want to chown every single file with user homedir?

Legit question, I am trying to think situations where user would not want to have other files changed behind their back.
One I can think of is suid/sgid.

Reply:
Yeah, I started thinking more, recursive here is a bad idea.